src/libgpu/src/glsl2/glsl2_translate.cpp

```diff
@@ -586,9 +586,9 @@ insertFileHeader(State &state)
       }
    }
 
-   // Samplers
-   for (auto id = 0u; id < state.shader->samplerDim.size(); ++id) {
-      auto dim = state.shader->samplerDim[id];
+   // Samplers,use the number of samplerUsage for cycle
+   for (auto id = 0u; id < state.shader->samplerUsage.size(); ++id) {
+      auto dim = state.shader->samplerDim[id%state.shader->samplerDim.size()];
```

Comment:
* Could extract `state.shader->samplerDim.size()` to a new `const` variable.
* Might want to space things out when using the module operator, like with all operators.